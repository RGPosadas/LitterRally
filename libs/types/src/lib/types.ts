// Types go here

export interface Message {
  message: string;
  info: { production: boolean };
}

export interface IMedia {
  id: number;
  mediaType: string;
}

/**
 * Restriction options for events
 */
export enum EventRestriction {
  NONE,
  FRIENDS_ONLY,
  INVITE_ONLY,
}
