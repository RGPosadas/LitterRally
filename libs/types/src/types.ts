// Types go here
import { Request, Response } from 'express';
export interface Message {
  message: string;
  info: { production: boolean };
}

export interface IMedia {
  id: number;
  mediaType: string;
}

/**
 * Restriction options for events
 */
export enum EventRestriction {
  NONE,
  FRIENDS_ONLY,
  INVITE_ONLY,
}

export enum RegistrationMethod {
  LOCAL = 'LOCAL',
  GOOGLE = 'GOOGLE',
  FACEBOOK = 'FACEBOOK',
  TWITTER = 'TWITTER',
}

export enum RelationshipType {
  FRIENDS,
  ADDED_ME,
  PENDING_REQUEST,
  NONE,
}

export const EventRestrictionMap = ['Everyone', 'Friends', 'Invite Only'];

export interface IEvent {
  id: number;
  title: string;
  startDate: Date;
  endDate: Date;
  description: string;
  image?: IMedia;
  restriction: EventRestriction;
  participants?: IUser[];
  host?: IUser;
  coHosts?: IUser[];
  location?: ILocation;
}

export interface ISerializedEvent extends Omit<IEvent, 'startDate' | 'endDate'> {
  startDate: string;
  endDate: string;
}

export interface ILocation {
  id?: number;
  title: string;
  event?: IEvent;
  coordinates?: IPoint;
  placeId?: string;
}

export interface IPoint {
  lat: number;
  long: number;
}

export interface IUser {
  id: number;
  name?: string;
}

export interface IMedia {
  id: number;
  mediaType: string;
}

export interface IAuthToken {
  id: number;
  iat: number;
  exp: number;
}

export type GqlContext = {
  req: Request;
  res: Response;
};

export interface IRefreshResponse {
  ok: boolean;
  accessToken: string;
}

export interface IRoute {
  url: string;
  displayName: string;
}

export class IGooglePlace {
  description: string;
  placeId: string;
}

export interface IWebSocketParams {
  authToken: string;
}

export interface IPost {
  id: number;
  user: IUser;
  parentPost: IPost;
  message: string;
  createdTime: Date;
  medias?: IMedia;
  channel: IChannel;
}

export interface ISerializedPost extends Omit<IPost, 'createdTime'> {
  createdTime: string;
}

export interface IChannel {
  id: number;
  name: string;
  rights: number;
  event?: ISerializedEvent;
  participants: IUser[];
  posts: ISerializedPost[];
}
