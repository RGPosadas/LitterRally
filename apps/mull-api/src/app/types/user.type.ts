import { Field, ID, ObjectType } from '@nestjs/graphql';

@ObjectType('User')
export class UserType {
  @Field(/* istanbul ignore next */ () => ID)
  id: number;

  @Field()
  password: string;

  @Field()
  email: string;

  @Field()
  timezone: string;

  @Field()
  name: string;

  @Field()
  age: string;

  @Field()
  description: string;

  @Field(/* istanbul ignore next */ () => [UserType])
  friends: UserType[];

  // TODO: Implement modules, services, types for Entities below

  //   @Field(() => [EventType])
  //   events: EventType[];

  //   @Field(() => [PostReactionType])
  //   postReactions: PostReactionType[];

  //   @Field(() => MediaType)
  //   avatar: MediaType;
}
