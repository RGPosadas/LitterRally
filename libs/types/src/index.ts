export * from './forms';
export * from './host.env';
export * from './ml';
export * from './types';
