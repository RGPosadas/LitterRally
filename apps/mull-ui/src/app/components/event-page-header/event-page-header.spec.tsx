import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

import EventPageHeader from './event-page-header';
import { IEvent } from '@mull/types';

describe('EventPageHeader', () => {
  const dummyEvent: IEvent = {
    id: 1,
    title: 'Test title',
    description: 'Test description',
    startDate: new Date(2000, 0, 1, 0, 0, 0, 0),
    endDate: new Date(2000, 0, 2, 0, 0, 0, 0),
  };

  it('should render successfully', () => {
    const { baseElement } = render(<EventPageHeader event={dummyEvent} />);
    expect(baseElement).toBeTruthy();
  });

  it('should match snapshot', () => {
    const tree = renderer.create(<EventPageHeader event={dummyEvent} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render dates correctly', () => {
    const dom = render(<EventPageHeader event={dummyEvent} />);

    const startDateDiv = dom.getByTestId('start-date-div');
    const endDateDiv = dom.getByTestId('end-date-div');
    expect(startDateDiv.textContent).toBe('Jan 1\n12:00 AM');
    expect(endDateDiv.textContent).toBe('Jan 2\n12:00 AM');
  });
});
