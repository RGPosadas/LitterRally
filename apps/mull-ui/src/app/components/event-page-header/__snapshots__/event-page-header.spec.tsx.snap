// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventPageHeader should match snapshot 1`] = `
<div
  className="event-page-header"
>
  <button
    className="mull-back-button undefined"
    data-testid="mull-back-button"
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-chevron-left fa-w-10 mull-back-button-icon"
      data-icon="chevron-left"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 320 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <div
      className="mull-back-button-text"
    >
      
    </div>
  </button>
  <div
    className="title"
  >
    Test title
  </div>
  <img
    alt="Event Page"
    className="event-image"
    src=""
  />
  <div
    className="event-datetime"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-clock fa-w-16 clock-icon event-page-icon color-grey"
      data-icon="clock"
      data-prefix="far"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200zm61.8-104.4l-84.9-61.7c-3.1-2.3-4.9-5.9-4.9-9.7V116c0-6.6 5.4-12 12-12h32c6.6 0 12 5.4 12 12v141.7l66.8 48.6c5.4 3.9 6.5 11.4 2.6 16.8L334.6 349c-3.9 5.3-11.4 6.5-16.8 2.6z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <div
      className="event-datetime-string"
      data-testid="start-date-div"
    >
      <div>
        1 Jan
      </div>
      <div>
        12:00 AM
      </div>
    </div>
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-caret-right fa-w-6 event-page-icon color-grey"
      data-icon="caret-right"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 192 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <div
      className="event-datetime-string"
      data-testid="end-date-div"
    >
      <div>
        2 Jan
      </div>
      <div>
        12:00 AM
      </div>
    </div>
  </div>
</div>
`;
